### PR TITLE
[YAML] Document jinja templatization features.

### LINKED ISSUE #32873: [Task]: Document Jinja templatization for yaml
### What needs to happen?

This is mentioned nowhere in https://beam.apache.org/documentation/sdks/yaml/ or our other yaml docs. 

### Issue Priority

Priority: 2 (default / most normal work should be filed as P2)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [X] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [X] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by robertwb**
This resolves #32873.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @Polber

File: website/www/site/content/en/documentation/sdks/yaml.md [hunk 1]
```diff
@@ -708,6 +708,49 @@ options:
   streaming: true
 ```
 
+## Jinja Templatization
+
+It is a common to want to run a single Beam pipeline in different contexts
+and/or with different configurations.
+When running a YAML pipeline either locally or via the gcloud,
```
Comment (mravi) [inline website/www/site/content/en/documentation/sdks/yaml.md:715]: nit: should we re-word to say `locally or against a remote pipeline runner` ? as gcloud seem restrictive.

Comment (robertwb) [reply website/www/site/content/en/documentation/sdks/yaml.md:715]: Yeah, I agree this isn't worded well. It's not really about the runner that's used, but the entrypoint. (E.g. we don't support this for embedded YamlTransform.)

Clarified that it's about both `apache_beam.yaml.main` (which an be local or various remote runners) and `gcloud` (which at the end of the day is `apache_beam.yaml.main`, but isn't s obvious and doesn't require beam to be installed).

Comment (mravi) [reply website/www/site/content/en/documentation/sdks/yaml.md]: ty for the clarification!

**Review by mravi — APPROVED** [review]

**Comment by robertwb** [comment]
Thanks!

**Pull request merged**